extensions/amp-autocomplete/0.1/test/validator-amp-autocomplete.out

```diff
@@ -0,0 +1,183 @@
+|          { items: [ "a", "b", "c" ] }
+|        </script>
+|    </amp-autocomplete>
+|
+|    <!-- Valid: amp-component with static data and min-characters attribute -->
+|    <amp-autocomplete min-characters=3>
+|        <input type="text">
+|        <script type="application/json">
+|          { items: [ "a", "b", "c" ] }
+|        </script>
+|    </amp-autocomplete>
+|
+|    <!-- Valid: amp-component with static data and max-entries attribute -->
+|    <amp-autocomplete max-entries=10>
+|        <input type="text">
+|        <script type="application/json">
+|          { items: [ "a", "b", "c" ] }
+|        </script>
+|    </amp-autocomplete>
+|
+|    <!-- Valid: amp-component with static data and suggest-first attribute -->
+|    <amp-autocomplete suggest-first>
+|        <input type="text">
+|        <script type="application/json">
+|          { items: [ "a", "b", "c" ] }
+|        </script>
+|    </amp-autocomplete>  
+|
+|    <!-- Valid: amp-component with static data and submit-on-enter attribute -->
+|    <amp-autocomplete submit-on-enter>
+|        <input type="text">
+|        <script type="application/json">
+|          { items: [ "a", "b", "c" ] }
+|        </script>
+|    </amp-autocomplete> 
+|    
+|    <!-- Valid: input tag only (no data source) -->
+|    <amp-autocomplete>
+|      <input type="text">
+|    </amp-autocomplete>  
+|
+|    <!-- Valid: input tag with type="search" -->
+|    <amp-autocomplete>
+|      <input type="search">
+|    </amp-autocomplete>  
+|
+|    <!-- Invalid: width is mistyped -->
+|    <amp-autocomplete widht=100>
+>>   ^~~~~~~~~
+amp-autocomplete/0.1/test/validator-amp-autocomplete.html:126:2 The attribute 'widht' may not appear in tag 'amp-autocomplete'. (see https://www.ampproject.org/docs/reference/components/amp-autocomplete) [DISALLOWED_HTML]
+|        <input type="text">
+|        <script type="application/json">
+|          { items: [ "a", "b", "c" ] }
+|        </script>
+|    </amp-autocomplete>  
+|
+|    <!-- Invalid: responsive layout is not supported -->
+|    <amp-autocomplete layout="responsive">
+>>   ^~~~~~~~~
+amp-autocomplete/0.1/test/validator-amp-autocomplete.html:134:2 The specified layout 'RESPONSIVE' is not supported by tag 'amp-autocomplete'. (see https://www.ampproject.org/docs/reference/components/amp-autocomplete) [AMP_LAYOUT_PROBLEM]
+|        <input type="text">
+|        <script type="application/json">
+|          { items: [ "a", "b", "c" ] }
+|        </script>
+|    </amp-autocomplete>  
+|
+|    <!-- Invalid: no children -->
+|    <amp-autocomplete>
+>>   ^~~~~~~~~
+amp-autocomplete/0.1/test/validator-amp-autocomplete.html:142:2 Tag 'amp-autocomplete' must have a minimum of 1 child tags - saw 0 child tags. (see https://www.ampproject.org/docs/reference/components/amp-autocomplete) [GENERIC]
+|    </amp-autocomplete>  
+|    
+|    <!-- Invalid: script tag present with input tag excluded -->
+|    <amp-autocomplete>
+|      <script type="application/json">
```

Comment:
This tag didn't get marked as invalid

Reply:
I asked in an earlier thread how I can better specify this type of rule so it does get marked invalid. `<input>` should always be a child of autocomplete and `<script>` can sometimes be, but I'm not sure how to specify one as always mandatory and the other as optional. From the convention I am using it makes it mandatory to have at least one of the two but it doesn't matter which.

Reply:
Ah sorry, I can't find it. GitHub can make it hard to find comment threads after changes have been made on top : / I'll take a look

Reply:
As we discussed, `unique` on the reference points and `mandatory` on the input reference point makes this invalid as required, and you can omit the `child_tags` spec

Reply:
No worries, thanks for your help! This should be all set now.